wp-content/plugins/core/src/Blocks/Block_Base.php

```diff
@@ -83,7 +84,20 @@ public function enqueue_core_block_public_styles(): void {
 			return;
 		}
 
-		wp_add_inline_style( $handle, file_get_contents( $src_path ) );
+		if ( $this->global_enqueue ) {
+			$args = $this->get_asset_file_args( get_theme_file_path( "dist/blocks/$path/index.asset.php" ) );
+
+			wp_enqueue_style(
+				$this->get_block_handle(),
+				$src,
+				$args['dependencies'] ?? [],
+				$args['version'] ?? false,
+				'all'
+			);
+		} else {
+			wp_add_inline_style( $handle, file_get_contents( $src_path ) );
+		}
+
 		add_editor_style( $src );
```

Comment:
A couple suggestions, if you'd be so kind:
```suggestion
		// Add the style in the editor
		add_editor_style( $src );

		if ( $this->global_enqueue ) {
			// Move `$args = ....` line to the top of the method with the other property definitions.
			// Even though it is only used conditionally, it's better to declare all setup vars in the same location.)
			// $args = $this->get_asset_file_args( get_theme_file_path( "dist/blocks/$path/index.asset.php" ) );
			// Enqueue the style unconditionally on the public site
			wp_enqueue_style(
				$this->get_block_handle(),
				$src,
				$args['dependencies'] ?? [],
				$args['version'] ?? false,
				'all'
			);
				
			return;
		}

		// Conditionally inline the style for the public site
		wp_add_inline_style( $handle, file_get_contents( $src_path ) );

```

Reply:
Sorry for the confusion. My first question is not related to styled insert into the editor. It's specific to core block styles being rendered on the frontend if the block in question was not used in the block editor (post_content). For example, If I were to add a columns block (or use the columns block classes) in a site footer template--but not in the current posts's content--would WordPress's core styles for that columns block be inserted on the frontend or not?

I ask this because the `wp_add_inline_style()` is designed to attach additional (our custom) styles to the existing  styles for a block. So, as long as the core block styles are being rendered on the FE, then our custom styles should also be rendered.

On the flip side, if the core styles are _not_ being rendered on the front end for blocks that are not used in the post_content, then isn't there a chance that the styling of that block (columns in this case) will be incorrect?

Reply:
I see what you're saying. I think the main issue we're trying to solve here is for the button block, which, for the most part, doesn't get affected by the blocks' core styles being missing because of how heavily we've overridden the block with our own styles in `theme.json` & custom block styles. I can see where issues would creep in if you're trying to use this on other blocks.

I'm willing to hold off on this PR for the next project to see if we run into this issue again and maybe we can come up with another solve. 

The specific example from comic-con was that they wanted to be able to style form submits & a few other unrelated button type things the same as the core buttons. They just ended up duplicating those styles but maybe that's how it _should_ be done and we leave core blocks alone to exist as their own thing? Maybe just be a bit better at documenting in-code if there are similar code blocks elsewhere?